src/main/java/kitchenpos/menu/domain/MenuProducts.java

```diff
@@ -0,0 +1,34 @@
+package kitchenpos.menu.domain;
+
+import kitchenpos.common.domain.Price;
+
+import javax.persistence.*;
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+@Embeddable
+public class MenuProducts {
+    @OneToMany(mappedBy = "menu", fetch = FetchType.LAZY, cascade = CascadeType.PERSIST, orphanRemoval = true)
+    private List<MenuProduct> menuProducts = new ArrayList<>();
+
+    protected MenuProducts() {}
+
+    public MenuProducts(List<MenuProduct> menuProducts) {
+        this.menuProducts = new ArrayList<>(menuProducts);
+    }
+
+    public Price totalMenuPrice() {
+        Price total = new Price(BigDecimal.ZERO);
+        for (MenuProduct menuProduct : menuProducts) {
+            total = total.add(menuProduct.calculatePrice());
+        }
+
+        return total;
+    }
```

Comment:
지금 구현해주신 방법도 좋지만 
`stream` 의 `reduce` 를 활용하는 방법도 있으니 참고만 해주세요!! 🙇‍♂️
```suggestion
    public Price totalMenuPrice() {
        return menuProducts.stream()
                .map(MenuProduct::calculatePrice)
                .reduce(new Price(BigDecimal.ZERO), Price::add);
    }
```